cumulus_etl/deid/scrubber.py

```diff
@@ -406,23 +406,31 @@ def _check_text(self, key: str, value: Any) -> Any:
         if isinstance(value, str) and any(
             (
                 # Coding.display is clinically unnecessary but is useful for rendering.
-                # Since "code" is always present, downstream consumers can & should provide their own display label.
-                # But we don't remove it entirely, for cases where unexpected codes are used.
-                # Note that this will definitely over-scrub (like scrubbing "White" from the USCDI race extension),
-                # but again -- this display value is redundant and rather than try to be smart, we're safely dumb.
+                # Since "code" is always present, downstream consumers can & should provide their
+                # own display label. But we don't remove it entirely, for cases where unexpected
+                # codes are used. Note that this will definitely over-scrub (like scrubbing "White"
+                # from the USCDI race extension), but again -- this display value is redundant and
+                # rather than try to be smart, we're safely dumb.
                 key == "display",
-                # CodeableConcept.text has clinical value for situations that don't have clear coding yet.
-                # Think early-days Covid day PCRs. Which is why we let it through in the first place.
+                # CodeableConcept.text has clinical value for situations that don't have clear
+                # coding yet, like early-days Covid day PCRs. And text-only codeable concepts show
+                # up a lot when the EHR allows it. Hence why we normally let it through.
                 # But we should still scrub it since it is loose text that could hold PHI.
                 key == "text",
-                # Observation.valueString has clinical value, but could hold PHI.
-                # Similarly, extensions might have valueString members (though the only supported ones don't have
-                # interesting fields there -- race & ethnicity allow for freeform text descriptions).
-                key == "valueString",
             )
         ):
             value = self.scrub_text(value)
 
+        if isinstance(value, str) and any(
+            (
+                # Observation.valueString has some clinical value, but often holds PHI.
+                # If we ever want to use this for clinical purposes, we should process it via NLP
+                # on the ETL side (like we do for DocumentReference attachments).
+                key == "valueString",
+            )
+        ):
+            raise MaskValue
```

Comment:
i'm sure this is correct, but just for my own edification, where is this caught?

Reply:
Fair question :smile: 

The loop of the scrubber is that it walks the JSON tree, and runs all these little scrubbing handlers that if they recognize something they should scrub, modify the value or raise an exception like this.

So the answer is: one layer up in the Scrubber code. See https://github.com/smart-on-fhir/cumulus-etl/blob/741fb34678425759df58b91e6ba4da433f74e509/cumulus_etl/deid/scrubber.py#L167